src/components/LanguageToggle.js

```diff
@@ -1,24 +1,53 @@
 import React, { useContext } from 'react';
 import { MdLanguage } from 'react-icons/md';
 import { AppContext, Types } from '../context';
+import LOCALES from '../i18n/locales';
+import styled from 'styled-components';
+
+const Dropdown = styled.div`
+  font: 400 16px system-ui;
+  &:hover {
+    .dropdown-menu {
+      display: block;
+    }
+  }
+`;
 
 function LanguageToggle() {
   const { lang, dispatch } = useContext(AppContext);
-  const handleChange = () => {
+  const languages = Object.values(LOCALES);
+
+  //getting the current language displayName based on it's ID
+  const currentLang =
+    languages.filter((l) => l.id === lang)[0].displayName || 'fr-fr';
```

Comment:
find is better than filter ;)

Reply:
True that, will be replaced in upcoming PRs :)